### ISSUE TITLE
[java] UnusedAssignment: False negative when initializer is overwritten inside unconditional if(true) block

### ISSUE THREAD

**Issue by leemeii** [pmd/pmd#6518]
**Affects PMD Version:** 7.22.0

**Rule:**[UnusedAssignment
](https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#unusedassignment)

**Description:**
The UnusedAssignment rule  fails to trigger on the pattern when the overwriting assignment is inside an unconditional  ```if(true)``` block.
This is a clear false negative — the initializer ```null``` is never used in all execution paths, yet only the direct assignment case is flagged.

**Code Sample demonstrating the issue:**

```java
public class a{
    public void test1(){
        Object target=null;           // FN: should report 
        if(true){
            target = new Object();
        }   
        System.out.println(target);
    }
    public void test2(){
        Object target=null;           // TP: correctly reported
        
            target = new Object();
          
        System.out.println(target);
    }
}
```

**Expected outcome:**
PMD should report UnusedAssignment for the line ```Object target=null;``` in both ```test1()``` and ```test2()``` with the message:
```"The initializer for variable 'target' is never used (overwritten on line X)"```
But currently it only reports the direct case (TP), completely **missing the if(true) case (FN).**

**Note**: The UnconditionalIfStatement rule correctly flags the ```if(true)``` in test1(), but this information is not propagated to UnusedAssignment.

**Running PMD through:** *CLI*
